maestro-ios-driver/src/main/kotlin/xcuitest/XCTestDriverClient.kt

```diff
@@ -68,40 +69,32 @@ class XCTestDriverClient(
     private val mapper = jacksonObjectMapper()
 
     fun viewHierarchy(installedApps: Set<String>): ViewHierarchy {
-        return executeJsonRequest(
+        val responseString = executeJsonRequest(
             "viewHierarchy",
-            ViewHierarchyRequest(installedApps),
-            ViewHierarchy::class
+            ViewHierarchyRequest(installedApps)
         )
+        return mapper.readValue(responseString, ViewHierarchy::class.java)
```

Comment:
Should this be wrapped in a try/catch in the event of the parsing failing?

Reply:
If this fails we would know it because of the type of exception thrown and would lead to cancellation. Are you recommending to map error to something else